test/node/regressions/many-request-handlers.test.ts

```diff
@@ -1,58 +1,98 @@
 /**
  * @vitest-environment node
  */
+import { HttpServer } from '@open-draft/test-server/http'
 import { graphql, http, HttpResponse } from 'msw'
 import { setupServer } from 'msw/node'
+import * as nodeEvents from 'node:events'
 
-// Create a large number of request handlers.
-const restHandlers = new Array(100).fill(null).map((_, index) => {
-  return http.post(
-    `https://example.com/resource/${index}`,
-    async ({ request }) => {
-      const text = await request.text()
-      return HttpResponse.text(text + index.toString())
-    },
-  )
+const httpServer = new HttpServer((app) => {
+  app.post('/graphql', (_, res) => {
+    return res.status(500).send('original-response')
+  })
 })
 
-const graphqlHanlers = new Array(100).fill(null).map((_, index) => {
-  return graphql.query(`Get${index}`, () => {
-    return HttpResponse.json({ data: { index } })
-  })
+vi.mock('node:events', async () => {
+  const actual = (await vi.importActual('node:events')) as import('node:events')
+  return {
+    ...actual,
+    setMaxListeners: vi.fn(actual.setMaxListeners),
+  }
 })
 
-const server = setupServer(...restHandlers, ...graphqlHanlers)
+let server: ReturnType<typeof setupServer>
 
-beforeAll(() => {
+const spy = vi.mocked(nodeEvents.setMaxListeners)
+
+beforeAll(async () => {
+  await httpServer.listen()
+
+  // Create a large number of request handlers.
+  const restHandlers = new Array(100).fill(null).map((_, index) => {
+    return http.post(
+      httpServer.http.url(`/resource/${index}`),
+      async ({ request }) => {
+        const text = await request.text()
+        return HttpResponse.text(text + index.toString())
+      },
+    )
+  })
+
+  const graphqlHandlers = new Array(100).fill(null).map((_, index) => {
+    return graphql.query(`Get${index}`, () => {
+      return HttpResponse.json({ data: { index } })
+    })
+  })
+
+  server = setupServer(...restHandlers, ...graphqlHandlers)
   server.listen()
   vi.spyOn(process.stderr, 'write')
 })
 
-afterAll(() => {
+afterAll(async () => {
   server.close()
   vi.restoreAllMocks()
+  await httpServer.close()
 })
 
 it('does not print a memory leak warning when having many request handlers', async () => {
-  const httpResponse = await fetch('https://example.com/resource/42', {
+  const httpResponse = await fetch(`${httpServer.http.url(`/resource/99`)}`, {
     method: 'POST',
     body: 'request-body-',
   }).then((response) => response.text())
+  expect(spy).toHaveBeenNthCalledWith(1, 200, expect.any(AbortSignal))
 
-  const graphqlResponse = await fetch('https://example.com', {
+  const graphqlResponse = await fetch(httpServer.http.url('/graphql'), {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
     },
     body: JSON.stringify({
-      query: `query Get42 { index }`,
+      query: `query Get99 { index }`,
     }),
   }).then((response) => response.json())
 
+  expect(spy).toHaveBeenNthCalledWith(2, 200, expect.any(AbortSignal))
```

Comment:
we fail here with just requesting 99 because we're always at a minimum `n+1` listeners.  1 per handler + 1 per request. 

then we also add one per parse of grapqhl, and we're way over the limit!

Note - the HTTP call above works because order matters. the errors here would flip if we defined the gql handlers first